src/main/java/org/opentripplanner/routing/api/request/request/filter/TransitFilterRequest.java

```diff
@@ -2,50 +2,33 @@
 
 import java.io.Serializable;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
 import org.opentripplanner.framework.tostring.ToStringBuilder;
-import org.opentripplanner.transit.model.network.Route;
+import org.opentripplanner.transit.model.network.TripPattern;
 import org.opentripplanner.transit.model.timetable.TripTimes;
 
 public class TransitFilterRequest implements Serializable, TransitFilter {
 
-  public static class Builder {
-
-    private final List<SelectRequest> select = new ArrayList<>();
-    private final List<SelectRequest> not = new ArrayList<>();
+  /**
+   * This is stored as an array, as they are iterated over for each trip when filtering transit
+   * data. Iterator creation is relatively expensive compared to iterating over a short array.
+   */
+  private final SelectRequest[] select;
 
-    public Builder addSelect(SelectRequest selectRequest) {
-      this.select.add(selectRequest);
-
-      return this;
-    }
-
-    public Builder addNot(SelectRequest selectRequest) {
-      this.not.add(selectRequest);
-
-      return this;
-    }
-
-    public TransitFilterRequest build() {
-      return new TransitFilterRequest(this);
-    }
-  }
+  /**
+   * {@link TransitFilterRequest#select}
+   */
+  private final SelectRequest[] not;
 
   public TransitFilterRequest(Builder builder) {
-    this.select = Collections.unmodifiableList(builder.select);
-    this.not = Collections.unmodifiableList(builder.not);
+    this.select = builder.select.toArray(SelectRequest[]::new);
+    this.not = builder.not.toArray(SelectRequest[]::new);
   }
 
-  private final List<SelectRequest> select;
-  private final List<SelectRequest> not;
-
   public List<SelectRequest> select() {
-    return select;
-  }
-
-  public List<SelectRequest> not() {
-    return not;
+    return Collections.unmodifiableList(Arrays.asList(select));
```

Comment:
I think this should be `not`

Reply:
No, we no longer expose `not`, as it did have no usage.